src/main/scala/exercises/06_Collections.scala

```diff
@@ -165,30 +178,14 @@ object Collections {
    * The idea is to rewrite the method into more functional style. 
    */
   def separateTheYoungFromTheOld(persons: List[Person]): List[List[String]] = {
-    var youngins: ListBuffer[Person] = new ListBuffer[Person]()
-    var elders: ListBuffer[Person] = new ListBuffer[Person]()
-    var validYoungNames: ListBuffer[String] = new ListBuffer[String]()
-    var validOldNames: ListBuffer[String] = new ListBuffer[String]()
-
-    for (person <- persons) {
-        if (person.age < 18) {
-          youngins += person
-        } else {
-          elders += person
-        }
-    }
+    // seperate the young and old into two lists
+    val (young, old) = persons.partition(person => person.age < 18)
 
-    var sortedYoung = youngins.toList.sortBy(_.age)
-    var sortedOld = elders.toList.sortBy(_.age)
+    // sort both lists
+    val (sortedYoung, sortedOld) = (young.sortBy(_.age), old.sortBy(_.age))
 
-    for (young <- sortedYoung) {
-      validYoungNames += young.firstName
-    }
-    for (old <- sortedOld) {
-      validOldNames += old.firstName
-    }
-    
-    List(validYoungNames.toList, validOldNames.toList)
+    // return the list of just their names
+    List(sortedYoung.map(p => p.firstName), sortedOld.map(p => p.firstName))
```

Comment:
This looks nice. I also like that Scala can do method chaining, so this can be done in fewer lines as well. It really allows for the programmer to make choices based on their personal style.